samples/testapp/src/commonMain/kotlin/com/android/identity/testapp/ui/IsoMdocProximitySharingScreen.kt

```diff
@@ -93,171 +96,248 @@ fun IsoMdocProximitySharingScreen(
+                                contentDescription = null,
+                            )
+                        }
+                        IconButton(
+                            onClick = { settingsModel.swapNegotiatedHandoverOrder(n, n + 1) },
+                            enabled = !isLast
+                        ) {
+                            Icon(
+                                imageVector = Icons.Filled.ArrowDownward,
+                                contentDescription = null,
                             )
                         }
-                    },
-                    content = { Text("Share via QR (mdoc peripheral server mode w/ L2CAP)") }
+                    }
+                }
+            }
+            item {
+                SettingHeadline("Transport Options")
+            }
+            item {
+                SettingToggle(
+                    title = "Use L2CAP if available",
+                    isChecked = settingsModel.presentmentBleL2CapEnabled.collectAsState().value,
+                    onCheckedChange = { settingsModel.presentmentBleL2CapEnabled.value = it },
                 )
             }
+            item {
+                SettingToggle(
+                    title = "Keep connection open after first request",
+                    isChecked = settingsModel.presentmentAllowMultipleRequests.collectAsState().value,
+                    onCheckedChange = { settingsModel.presentmentAllowMultipleRequests.value = it },
+                )
+            }
+            item {
+                SettingHeadline("General Options")
+            }
+            item {
+                SettingToggle(
+                    title = "Skip consent prompt",
+                    isChecked = !settingsModel.presentmentShowConsentPrompt.collectAsState().value,
+                    onCheckedChange = { settingsModel.presentmentShowConsentPrompt.value = !it },
+                )
+            }
+            item {
+                HorizontalDivider(
+                    modifier = Modifier.padding(8.dp)
+                )
+            }
+            item {
+                Column(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    Button(
+                        onClick = {
+                            settingsModel.resetPresentmentSettings()
+                        },
+                    ) {
+                        Text(text = "Reset Settings")
+                    }
+                }
+            }
+            item {
+                HorizontalDivider(
+                    modifier = Modifier.padding(8.dp)
+                )
+            }
+            item {
+                Column(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    Button(
+                        onClick = {
```

Comment:
Lots of code for a button handler... New function?

Reply:
Same.